src/components/CalendarButton/CalendarButton.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import { START_DATE } from 'react-dates/constants';
+
+import { RangePickerController } from '../Calendar';
+import Button from '../Button';
+
+class CalendarButton extends Component {
+  state = { startDate: null, endDate: null, focusedInput: null };
+  buttonRef = null;
+
+  _onDatesChange = ({ startDate, endDate }) => {
+    this.setState({ startDate, endDate });
+
+    if (startDate && endDate) {
+      this.props.onDatesRangeChange({ startDate, endDate });
+    }
+  };
+
+  _onFocusChange = focusedInput => this.setState({ focusedInput });
+
+  _onButtonClick = () =>
+    this.setState(({ focusedInput }) => ({
+      focusedInput: focusedInput !== null ? null : START_DATE
+    }));
+
+  _currentSelection = () => {
```

Comment:
Is this being used as a computed prop? We haven't established this as a pattern, yet, but I've used actual getters for these somewhere in the Dashboard.

```javascript
get currentSelection() {
  /* calculate and return current selection string */
}
```

This way there is no need to execute the method in your JSX. You can just refer to `this.currentSelection`.

How do you feel about this. Would be something we could also put on the agenda for a patterns/style chapter meeting I want to plan for sometime soon.

Reply:
My initial gut feeling was getCurrentSelection. I would use `get property` when designing public api and would want to hide it from the user of the api. Within a codebase I would prefer to be explicit about it and the reader wont be needing to look at the definition in order to know that this is a function.